src/product.cpp

```diff
@@ -186,6 +186,16 @@ size_t tsym::Product::hash() const
     return std::hash<BasePtrList>{}(ops);
 }
 
+unsigned tsym::Product::complexity() const
+{
+    unsigned operandscomplexity = 0;
+
+     for (const auto& factors : ops)
```

Comment:
if you loop through a container, the running item should be named 'factor', not 'factors'.